File: Lesson_Java_07042025/app/src/main/java/com/example/lesson07042025/MainActivity.java
package com.example.lesson07042025;

import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;

import java.util.ArrayList;

// MainActivity.java
public class MainActivity extends AppCompatActivity implements TaskListFragment.TaskListListener, AddTaskFragment.AddTaskListener, TaskDetailFragment.TaskDetailListener {
    private ArrayList<Task> taskList = new ArrayList<>();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        showTaskList();
    }

    private void showTaskList() {
        getSupportFragmentManager().beginTransaction()
                .replace(R.id.fragment_container, TaskListFragment.newInstance(taskList))
                .commit();
    }

    @Override
    public void onAddTaskClicked() {
        getSupportFragmentManager().beginTransaction()
                .replace(R.id.fragment_container, new AddTaskFragment())
                .addToBackStack(null)
                .commit();
    }

    @Override
    public void onTaskSelected(int position) {
        getSupportFragmentManager().beginTransaction()
                .replace(R.id.fragment_container, TaskDetailFragment.newInstance(taskList.get(position), position))
                .addToBackStack(null)
                .commit();
    }

    @Override
    public void onTaskAdded(Task task) {
        taskList.add(task);
        getSupportFragmentManager().popBackStack();
        showTaskList();
    }

    @Override
    public void onTaskDeleted(int position) {
        taskList.remove(position);
        getSupportFragmentManager().popBackStack();
        showTaskList();
    }
}

File: HWLesson18/app/src/main/java/com/example/hwlesson18/MainActivity.kt
package com.example.hwlesson18

import android.os.Bundle
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import com.example.hwlesson18.R
import com.example.hwlesson18.model.City
import com.example.hwlesson18.model.Weather
import com.example.hwlesson18.ui.DetailsFragment

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        setContentView(R.layout.activity_main)

        if (savedInstanceState == null) {
            val weather = Weather(City("Алматы", 43.238949, 76.889709))
            supportFragmentManager.beginTransaction()
                .replace(R.id.container, DetailsFragment.newInstance(weather))
                .commit()
        }
    }
}
